### PR TITLE
Local header actions

### PR DESCRIPTION
**Pull request by teneightfive**
This change adds support for an _actions_ area on the right hand side of the local header component.

## Example

image

### CODE REVIEW

File: assets/stylesheets/_deprecated/components/_button.scss [hunk 1]
```diff
@@ -50,3 +50,8 @@
 .button--compact {
   padding: 0;
 }
+
+.button--small {
+  @include core-font(16);
```
Comment (feedmypixel) [inline assets/stylesheets/_deprecated/components/_button.scss]: why's this in deprecated? can it not be placed in non deprecated file?

File: assets/stylesheets/components/_local-header.scss [hunk 2]
```diff
@@ -49,6 +50,26 @@
   margin-bottom: 0;
 }
 
+.c-local-header__content {
+  @include media(tablet) {
+    float: left;
+    width: 70%;
```
Comment (feedmypixel) [inline assets/stylesheets/components/_local-header.scss]: not a massive fan of floats for layout but I feel that just personal preference. I prefer `display: table` and `display:table-cell`

File: src/templates/_macros/common.njk [hunk 3]
```diff
@@ -60,6 +60,7 @@
  # @param {string} [props.heading] - page heading
  # @param {string} [props.headingBefore] - a string before heading (safe to render HTML)
  # @param {string} [props.headingSuffix] - a string suffix for heading (safe to render HTML)
+ # @param {object} [props.actions] - a set of actions that will appear on the right hand side of the component
```
Comment (feedmypixel) [inline src/templates/_macros/common.njk]: maybe worth removing "that will appear on the right hand side of the component" as this is only true `> tablet`

File: assets/stylesheets/_deprecated/components/_button.scss (hunk 1 above)
Comment (teneightfive) [reply assets/stylesheets/_deprecated/components/_button.scss]: My guess is its in deprecated when we pulled the tradeelements dependency in. This however hasn't been refactored yet but there have been cases where we've needed to extend the button styles slightly.

I opted for this rather than refactor all of the buttons across the app for now. We could move it out and use BEM syntax as a separate change though.

File: assets/stylesheets/components/_local-header.scss (hunk 2 above)
Comment (teneightfive) [reply assets/stylesheets/components/_local-header.scss]: I gave it a go with tables and it feels like you need to mess around with more properties to get the same effect. You would need to apply `display: table` to the current `.l-container` `div` which would probably mean creating another element selector (`.c-local-header__container`) to apply that style.

Then you also have to ensure the element with display table has `width: 100%` and it started to feel more overloaded.

File: assets/stylesheets/_deprecated/components/_button.scss (hunk 1 above)
Comment (feedmypixel) [reply assets/stylesheets/_deprecated/components/_button.scss]: > We could move it out and use BEM syntax as a separate change though.

sounds good 👍

Comment (teneightfive) [reply assets/stylesheets/_deprecated/components/_button.scss]: I took a brief look into this at the weekend and realised it's a pretty big change 😞 

Basically because the current method is tied in to both the govuk_frontend_toolkit and govuk_elements so it would mean quite a bit of re-writing to remove both those deps. 

Or alternatively we could still keep the govuk_frontend_toolkit dependency but it felt like a half-way fix so didn't warrant the effort right now.

So in summary, I think we leave that until we get time to properly address the structure of the sass across the board.

Comment (feedmypixel) [reply assets/stylesheets/_deprecated/components/_button.scss]: Totally agreed - its one of the reasons that the button file is still in here. A lot of `govuk_frontend_toolkit` and `govuk_elements` are wound up with each other.

Something post November I feel
